File: backend/ParkingService.Domain/Entities/ViolationConfig.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ParkingService.Domain.Entities
{
    public class ViolationConfig
    {
        public Guid Id { get; set; }
        public string TypeName { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal FineAmount { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation
        public ICollection<Violation> Violations { get; set; } = new List<Violation>();
    
}
}

File: frontend/lib/features/parking/models/violation_config.dart
class ViolationConfig {
  final String id;
  final String typeName;
  final String description;
  final double fineAmount;
  final DateTime createdAt;
  final DateTime updatedAt;

  ViolationConfig({
    required this.id,
    required this.typeName,
    required this.description,
    required this.fineAmount,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ViolationConfig.fromJson(Map<String, dynamic> json) {
    return ViolationConfig(
      id: json['id'],
      typeName: json['typeName'],
      description: json['description'],
      fineAmount: (json['fineAmount'] as num).toDouble(),
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }
}